registry/slashFun.js

```diff
@@ -0,0 +1,163 @@
+// @ts-check
+const u = require('./regUtils');
+
+const roll = new u.sub()
+  .setName("roll")
+  .setDescription("Roll Dice")
+  .addIntegerOption(
+    new u.int()
+    .setName("dice")
+    .setDescription("How many dice to roll? (will roll at least 1 anyway) (max of 10000)")
+    .setRequired(false)
+    .setMinValue(1)
+    .setMaxValue(10000)
+  )
+  .addIntegerOption(
+    new u.int()
+    .setName("sides")
+    .setDescription("How many sides on the dice to roll? (defaults to 6)")
+    .setRequired(false)
+    .setMinValue(1)
+    .setMaxValue(2147483647)
+  )
+  .addIntegerOption(
+    new u.int()
+    .setName("modifier")
+    .setDescription("How much to change the roll by? (defaults to 0)")
+    .setRequired(false)
+  );
+const rollF = new u.sub()
+  .setName("rollf")
+  .setDescription("Roll Fate Dice")
+  .addIntegerOption(
+    new u.int()
+    .setName("dice")
+    .setDescription("How many dice to roll? (will roll at least 1 anyway) (max of 10000)")
+    .setRequired(false)
+    .setMinValue(1)
+    .setMaxValue(10000)
+  )
+  .addIntegerOption(
+    new u.int()
+    .setName("modifier")
+    .setDescription("How much to change the roll by? (defaults to 0)")
+    .setRequired(false)
+  );
+const rollOld = new u.sub()
+  .setName("rollold")
+  .setDescription("Roll Dice using the old formula format")
+  .addStringOption(
+    new u.string()
+    .setName("rollformula")
+    .setDescription("old dice formula to use, it errors in the same way as !roll did")
+  );
+const ball8 = new u.sub()
+  .setName("8ball")
+  .setDescription("Get an answer from the Magic 8-ball.")
+  .addStringOption(
+    new u.string()
+    .setName("question")
+    .setDescription("What do you wish to ask the 8-ball today?")
+    .setRequired(true)
+  );
+const allthe = new u.sub()
+  .setName("allthe")
+  .setDescription("ALL THE _____!")
+  .addStringOption(
+    new u.string()
+    .setName("thing")
+    .setDescription("something")
+    .setRequired(true)
+  );
+const repost = new u.sub()
+.setName("repost")
+.setDescription("That's a repost.");
```

Comment:
This one was scrapped a while back due to abuse

Reply:
I was there. I was the problem... now that it says _ used a /command on the replies it should be good. In fact, grey said he would fix it but never got to it.

Reply:
Also, I hit the wrong button and don't know how to un mark this if it isn't actually resolved and there was more specific problems with repost .

Reply:
I'm adamant to get this back in because I was directly involved with its removal and want to redeem myself. it hurts me to remove functionality from /roll, but I will allow that because I see the bloat. repost is less than 1/3 the size. at the very least, could I make a subsection called /fun image _ to remake all of the commands I ended up getting removed at that time?

Reply:
Ah ok. I took a look at the history behind the decision to remove it. I had forgotten about it being used to create a pseudo !say. iirc the purpose when it was created was because a lot of people in the memes channel were upset that there were a lot of frequent reposts. I'm not sure that we have that need anymore. Outside of that, I don't think there's much of a use for it. Feel free to correct me though, I'm a bit out of touch with the server as of late.

Reply:
I think it went out of use mainly because of the tag-over... I have a feeling with a new dedicated memes channel it may receive a decent uptick in usage again...